source/_posts/durable-entities.md

```diff
@@ -0,0 +1,129 @@
+    }
+    
+    [FunctionName(nameof(BedEntity))]
+    public static async Task HandleEntityOperation(
+        [EntityTrigger] IDurableEntityContext context,
+        ILogger logger)
+    {
+        await context.DispatchAsync<BedEntity>(logger, context.EntityKey);
+    }
+    
+    public Task<bool> IsOccupiedBedAsync()
+    {
+        var IsOccupied = BedNumber == "123";
+        return Task.FromResult(IsOccupied);
+    }
+    
+    public Task AssignBedAsync(string bedNumber)
+    {
+        IsOccupied = true;
+        return Task.CompletedTask;
+    }
+}
+```
+
+If we look at the above code you might be thinking but how and where does the call to the database take place to store the `BedEntity` after its modified? This is all abstracted by the framework. We do not need to worry about it. Both methods `IsOccupiedBedAsync` and `AssignBedAsync` change the state of the entity asynchronously and then the framework does the rest. The function `HandleEntityOperation` allows us to call operations on this entity from the client function or orchestrator.
+
+## Using the Entity
+
+Now let's have a look at a simple client function that would make use of this `BedEntity`:
+
+```csharp
+public static class BedManager
+{
+    [FunctionName(nameof(AssignBed))]
+    public static async Task<IActionResult> AssignBed(
+        [HttpTrigger(AuthorizationLevel.Function, "get")]
+        HttpRequest req,
+        [DurableClient] IDurableEntityClient durableEntityClient,
+        ILogger log)
+    {
+        var bedNumber = req.Query["bedNumber"];
+        var entityId = new EntityId(nameof(BedEntity),bedNumber);
+        
+        log.LogInformation($"Received request for bed {bedNumber}");
+        
+        var bedEntity = await durableEntityClient.ReadEntityStateAsync<BedEntity>(entityId);
+        if (bedEntity.EntityExists && bedEntity.EntityState.IsOccupied)
+        {
+            return new BadRequestObjectResult("Bed is already occupied.");
+        }
+        
+        await durableEntityClient.SignalEntityAsync(entityId, nameof(BedEntity.AssignBedAsync));
+
+        return new OkObjectResult($"Bed {bedNumber} has been set to occupied.");
+    }
+}
+```
+
+If we now run this client Function and send a request to it (bedNumber = 123) with a breakpoint on the logline we see the entityId being constructed as `@BedEntity@123`. The framework will use this id to get the current state if an entity is found using this id. If an entity is found using the `ReadEntityStateAsync` method on the `IDurableEntityClient` we check if the bed is also occupied. If that is the case we cannot assign it and return a `BadRequest`. If no BedEntity could be found with this id or the bed is not occupied we signal the entity to assign a bed with the provided `bedNumber` using the `SignalEntityAsync` method. This method gets the entity and triggers the `AssignBedAsync` call which changes the state. The framework then persists the changes in the entity state back to table storage.
```

Comment:
_If no BedEntity could be found with this id we signal the entity to assign a bed with the provided `bedNumber` using the `SignalEntityAsync` method._

ik zag het in de code maar notfound = assign?

Reply:
was een opsomming van opties / voorbeelden